phone/src/apps/sellout/hooks/useListing.ts

```diff
@@ -1,7 +1,12 @@
 import { useRecoilValue } from 'recoil';
+import { MarketplaceListing } from '../../../common/typings/marketplace';
 import { selloutState } from './state';
 
-export const useListing = () => {
-  const listing = useRecoilValue(selloutState.listing);
-  return { listing };
+interface ListingHook {
+  listings: MarketplaceListing[];
+}
+
+export const useListing = (): ListingHook => {
+  const listings = useRecoilValue(selloutState.listing);
+  return { listings };
```

Comment:
If you want to clean up this function a little bit so you don't need an interface/object property return, pass the useRecoilValue generic a type of MarketPlaceListing[], and simply return it.
```ts
export const useListing = () => {
  return useRecoilValue<MarketPlaceListing[]>(selloutState.listing)
}
```
Up to you but I think it looks a little cleaner this way

Reply:
Agree